lib/evss/service.rb

```diff
@@ -4,10 +4,24 @@
 
 module EVSS
   class Service < Common::Client::Base
+    STATSD_KEY_PREFIX = 'api.evss'
+
+    protected
+
     def headers_for_user(user)
       EVSS::AuthHeaders.new(user).to_h
     end
 
+    def with_monitoring
+      caller = caller_locations(1, 1)[0].label
+      yield
+    rescue StandardError => error
+      StatsD.increment("#{STATSD_KEY_PREFIX}.#{caller}.fail", tags: ["error:#{error.class}"])
+      handle_error(error)
```

Comment:
What about `raise error` instead?  `with_monitoring` is now responsible for both monitoring and error handling.  I think we can keep those responsibilities separate:
```ruby
with_error_handling do
  with_monitoring do
    # get letters
  end
end
```

Reply:
Was struggling with the name for that one, it used to be called `with_exception_handling` but when the statsd stuff goes in there the name doesn't apply anymore. I've always called statsd 'metrics' so I thought 'monitoring' could apply to both. If we separate them incrementing the total would work but how would a failure be incremented? Don't think it would make it to with_monitoring block because the exception would already be raised.